src/main/java/com/google/jenkins/plugins/storage/AbstractUploadDescriptor.java

```diff
@@ -57,7 +57,10 @@ protected AbstractUploadDescriptor(Class<? extends AbstractUpload> clazz) {
    *     this descriptor.
    */
   public UploadModule getModule() {
```

Comment:
this should be synchronized to prevent race conditoins

Reply:
Made the module final again so race conditions are avoided.

Reply:
Synchronized is probably still called for. While making the field final will ensure it's only set once, there still is are concurrent order of ops which could result in a runtime error.

Reply:
Done